js/test/test.js

```diff
@@ -180,8 +180,10 @@ async function testSafe(methodName, exchange, ...args) {
     }
 }
 
-
-async function testSymbol (exchange, symbol) {
+async function runPublicTests (exchange, symbol) {
+    const market = exchange.market (symbol);
+    const isSpot = market['spot'];
+    const params = isSpot ? { 'type': 'spot', 'defaultType': 'spot' } : { 'type': 'swap', 'defaultType': 'swap' };
```

Comment:
@ttodua why both? `defaultType` should be enough

Reply:
Also if you do `exchange.options['type'] = 'swap';` before calling this method, why do we need params?

Reply:
iirc it was not enough, bcz there were many cases across exchanges, using only `type` (probably old implementations, like this: https://github.com/ccxt/ccxt/blob/master/js/okcoin.js#L1567). so, maybe it doesn't hurt having both here.

Reply:
> Also if you do `exchange.options['type'] = 'swap';` before calling this method, why do we need params? 

again - because of imperfect implementations (some older ones) across ccxt. 
you can again look at bigone example I've linked (and there are others like that) where `options['type']` doesn't change anything in specific (imperfect) implementations, that's why we need both `params` and `options` to have more robust provision. 
imo, that doesn't hurt anything, does it?

Reply:
I understand that 'we can trace down all such imperfections across all exchanges' but that will take away too much out of scope atm, and I want we could merge this one without reworking specific exchanges atm.

Reply:
> imo, that doesn't hurt anything, does it?

Yes it does, because not every method is prepared to receive `type` or `defaultType` in params. 

(btw adding some changes right now)

@ttodua